File: 0268-missing-number/0268-missing-number.go
/*
func missingNumber(nums []int) int {
    occurred := map[int]int{}

    for _, v := range nums {
        occurred[v]++
    }

    for i := 0; i < len(nums)+1; i++ {
        if _, ok := occurred[i]; !ok {
            return i
        }
    }

    return -1
}
*/

func missingNumber(nums []int) int {
    got := 0
    exp := len(nums)
    for i := 0; i < len(nums); i++ {
        got += nums[i]
        exp += i
    }
    return exp - got
}

File: 0268-missing-number/0268-missing-number.py
"""
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        occurred = {}

        for n in nums:
            occurred[n] = True

        for i in range(len(nums)+1):
            if not occurred.get(i):
                return i

        return None        
"""

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        sum_n = ((0 + len(nums)) / 2) * (len(nums) + 1)
        return int(sum_n) - sum(nums)
